src/stp/new_tactics/TestTactic.cpp

```diff
@@ -28,11 +28,29 @@ void TestTactic::onTerminate() noexcept {
 StpInfo TestTactic::calculateInfoForSkill(StpInfo const &info) noexcept {
     StpInfo skillStpInfo = info;
 
-    skillStpInfo.setAngle(2.0);
-    skillStpInfo.setDribblerSpeed(31);
-    if (!skillStpInfo.getPositionToMoveTo()) {
-        skillStpInfo.setPositionToMoveTo(Vector2(0,0));
+    auto length = skillStpInfo.getField()->getFieldLength();
+    auto width = skillStpInfo.getField()->getFieldWidth();
+
+    switch (skills.current_num()) {
+    case 0:
+        skillStpInfo.setPositionToMoveTo(Vector2(length / 8, width / 8));
+        break;
+    case 1:
+        skillStpInfo.setPositionToMoveTo(Vector2(length / 8, -width / 8));
+        break;
+    case 2:
+        skillStpInfo.setPositionToMoveTo(Vector2(-length / 8, -width / 8));
+        break;
+    case 3:
+        skillStpInfo.setPositionToMoveTo(Vector2(-length / 8, width / 8));
+        break;
+    case 4:
+        skillStpInfo.setPositionToMoveTo(Vector2(length / 8, width / 8));
```

Comment:
Like I said, position should not get encoded this way. Create a separate variable in the Field class for these position or create a function in the FieldComputation class that returns these positions instead.